tests/api/test_api.py

```diff
@@ -0,0 +1,85 @@
+import re
+import pytest
+import string
+import random
+
+from lib.api import fusor_api
+
+####################################################################################################
+# This contains test designed to verify parts of the fusor api functionality. There will be no full
+# deployments just builing up part of a deployment to test the functionality of individual api calls
+#
+# NOTE: There is some overlap in fixtures from test_api_osp_deployment because we need to refactor
+#       how fixtures are shared across tests.
+# TODO: Fix how we share fixtures across tests
+####################################################################################################
+
+
+def parse_ip_from_url(url):
+    """
+    Use a regular expression to parse the IP from a url.
+    Will return IP address if not protocol is specified
+    This assumes that the url does not use a DNS address
+    """
+    # Regex groups 1 and 2 will make up the protocol and separator
+    # Group 3 will be the IP address
+    # Group for will be everything after the IP
+    url_pattern = r'((https?)://)?(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})(.*)'
```

Comment:
It was a quick fix . I didn't really think to search for a library since it was a simple regex. I'd rather use that library in the long run so I'll switch to that in a follow up PR